datalad/tests/test_dataset.py

```diff
@@ -83,3 +95,84 @@ def test_Dataset_get(src, dst):
     assert_equal(f.readlines(), ['123\n'], "test-annex.dat's content doesn't match.")
 
     os.chdir(cwd)
+
+
+@assert_cwd_unchanged
+@with_testrepos(flavors=local_flavors)
+@with_tempfile
+def test_Dataset_add_to_annex(src, dst):
+
+    ds = Dataset(dst, src)
+    filename = 'file_to_annex.dat'
+    filename_abs = os.path.join(dst, filename)
+    f = open(filename_abs, 'w')
+    f.write("What to write?")
+    f.close()
+    cwd = os.getcwd()
+    os.chdir(dst)
```

Comment:
Yes, it should not be necessary, therefore there is no need for you to
change the assert_cwd_unchanged decorator. I'm working on unifying the way
paths are handled and then will adapt the tests. They will end up with no
need to cd.

2015-04-08 15:20 GMT+02:00 Yaroslav Halchenko <notifications@github.com>:

> In datalad/tests/test_dataset.py
> <https://github.com/datalad/datalad/pull/90#discussion_r27968197>:
>
> > @@ -83,3 +95,84 @@ def test_Dataset_get(src, dst):
> >      assert_equal(f.readlines(), ['123\n'], "test-annex.dat's content doesn't match.")
> >
> >      os.chdir(cwd)
> > +
> > +
> > +@assert_cwd_unchanged
> > +@with_testrepos(flavors=local_flavors)
> > +@with_tempfile
> > +def test_Dataset_add_to_annex(src, dst):
> > +
> > +    ds = Dataset(dst, src)
> > +    filename = 'file_to_annex.dat'
> > +    filename_abs = os.path.join(dst, filename)
> > +    f = open(filename_abs, 'w')
> > +    f.write("What to write?")
> > +    f.close()
> > +    cwd = os.getcwd()
> > +    os.chdir(dst)
>
> since Dataset knows where it is, there should be no need to chdir
> explicitly at all. We might have a test (or sweep in all) where we pass
> full/relative/change path, but in general -- should not be necessary
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/datalad/datalad/pull/90/files#r27968197>.
>